cmd/mimir/help-all.txt.tmpl

```diff
@@ -1389,6 +1389,8 @@ Usage of ./cmd/mimir/mimir:
     	[experimental] Whether to enable translation of OTel start timestamps to Prometheus zero samples in the OTLP endpoint.
   -distributor.otel-metric-suffixes-enabled
     	Whether to enable automatic suffixes to names of metrics ingested through OTLP.
+  -distributor.promote-otel-resource-attributes comma-separated-list-of-strings
```

Comment:
We're using both "otel" and "otlp". Can we pick one to standardize on and name the flags consistently? e.g. Should "otlp" or "otel" be the first part of the flag name? `otlp-some-feature` or `otel-some-feature`

Reply:
OTLP and OTel are not synonyms though. OTLP is OTel's wire protocol. OTLP resource attribute doesn't make sense.

Reply:
I don't think that's a distinction that matters. If we are using the OTLP we're using "OTEL" functionality. I think there is value into having all the OpenTelemetry functionality share some particular naming. Maybe a `distributor.otel.` prefix would be helpful.

> Equating OTel and OTLP would be like equating Prometheus and PRW

In the distributor, that seems entirely fair.

Reply:
I would find equating the two really confusing at least. OTLP resource attribute just wouldn't make sense.

Reply:
Arve's explanation was super clear to me.

To add a bit more to it in case you want to see official docs
- OTLP Stands for OpenTelemetry Protocol and here is the spec  https://opentelemetry.io/docs/specs/otlp/
- Resource attributes are just labels that belong to the entity for which the telemetry is being recorded https://opentelemetry.io/docs/specs/otel/overview/#resources

Note each thing has its own spec 🤯 
![image](https://github.com/user-attachments/assets/38f5112c-30b1-409b-b894-c08f0e08fb96)

Reply:
Fine. I'd still like a little consistency in how the flags are named e.g. the prefix or the position of "otel" or "otlp" within the flag name.

Reply:
I'll see whether the `-distributor.otel.` prefix can work.